R/read.R

```diff
@@ -289,23 +309,89 @@ readH5AD <- function(file, X_name = NULL, use_hdf5 = FALSE,
     mat
 }
 
+.convert_element <- function(obj, path, file, recursive=FALSE) {
+    # Should we wrap this in suppressWarnings? rhdf5 will warn that it
+    # can't yet read enum (factor/boolean) in attributes
+    element_attrs <- rhdf5::h5readAttributes(file, path)
+
+    # Convert categorical element
+    if (identical(element_attrs[["encoding-type"]], "categorical")) {
+        codes <- obj[["codes"]] + 1
+        codes[codes == 0] <- NA
+        levels <- obj[["categories"]]
+
+        # Can't determine orderedness due to rhdf5 not yet supporting
+        # enums in attributes
+        #ord <- as.logical(element_attrs[["ordered"]])
```

Comment:
Ah, I guess this is where the warning comes in. If we aren't really addressing it maybe we should leave it (and try to fix this upstream).

Reply:
Agree it would be good to fix this upstream...

Also, I consolidated the 2 comments into a single comment, to try and make this issue easier to keep track of.